packages/contracts-bedrock/contracts/L2/L2StandardBridge.sol

```diff
@@ -68,6 +68,27 @@ contract L2StandardBridge is StandardBridge, Semver {
         StandardBridge(payable(Predeploys.L2_CROSS_DOMAIN_MESSENGER), _otherBridge)
     {}
 
+    /**
+     * @notice Allows EOAs to withdraw ETH by sending directly to the bridge.
+     *
+     * @dev Emits both an `WithdrawalInitiated` and `ETHBridgeInitiated` event.
+     */
+    receive() external payable override onlyEOA {
+        bytes calldata b;
+        assembly {
+            b.offset := 0
+            b.length := 0
+        }
+        _initiateWithdrawal(
+            Predeploys.LEGACY_ERC20_ETH,
+            msg.sender,
+            msg.sender,
+            msg.value,
+            RECEIVE_DEFAULT_GAS_LIMIT,
+            b
```

Comment:
Could we get rid of the assembly by passing through `msg.data`? I know we don't do that elsewhere and I think the data will just end up in an event

Reply:
Ah whoops missed this one, yes